device-code-flow-console/PublicAppUsingDeviceCodeFlow.cs

```diff
@@ -0,0 +1,178 @@
+            {
+                // Kind of errors you could have (in errorCode and ex.Message)
+                string errorCode = ex.ErrorCode;
+
+                // AADSTS50059: No tenant-identifying information found in either the request or implied by any provided credentials.
+                // Mitigation: as explained in the message from Azure AD, the authoriy needs to be tenanted. you have probably created
+                // your public client application with the following authorities:
+                // https://login.microsoftonline.com/common or https://login.microsoftonline.com/organizations
+
+                // AADSTS90133: Device Code flow is not supported under /common or /consumers endpoint.
+                // Mitigation: as explained in the message from Azure AD, the authority needs to be tenanted
+
+                // AADSTS90002: Tenant <tenantId or domain you used in the authority> not found. This may happen if there are 
+                // no active subscriptions for the tenant. Check with your subscription administrator.
+                // Mitigation: if you have an active subscription for the tenant this might be that you have a typo in the 
+                // tenantId (GUID) or tenant domain name, update the 
+
+                // The issues above are typically programming / app configuration errors, they need to be fixed
+                throw;
+            }
+
+            catch (OperationCanceledException)
+            {
+                // If you use an override with a CancellationToken, and call the Cancel() method on it, then this may be triggered
+                // to indicate that the operation was cancelled. 
+                // See https://docs.microsoft.com/en-us/dotnet/standard/threading/cancellation-in-managed-threads 
+                // for more detailed information on how C# supports cancellation in managed threads.
+                result = null;
+            }
+
+            catch (MsalClientException ex)
+            {
+                string errorCode = ex.ErrorCode;
+
+                // Verification code expired before contacting the server
+                // This exception will occur if the user does not manage to sign-in before a time out (15 mins) and the
+                // call to `AcquireTokenWithDeviceCodeAsync` is not cancelled in between
+                result = null;
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Calls the protected Web API and processes the result
+        /// </summary>
+        /// <param name="webApiUrl">Url of the Web API to call (supposed to return Json)</param>
+        /// <param name="accessToken">Access token used as a bearer security token to call the Web API</param>
+        /// <param name="processResult">Callback used to process the result of the call to the Web API</param>
+        public async Task CallWebApiAndDisplayResultASync(string webApiUrl, string accessToken, Action<JObject> processResult)
```

Comment:
If you are going to go with composition, I would move this method to MyInformation class

Reply:
Thanks. I've done it